packages/bpk-component-autosuggestV2/docs/migrating-from-v1-to-v2.md

```diff
@@ -0,0 +1,101 @@
+# Migrating from `BpkAutosuggest` to `BpkAutosuggestV2`
+
+`BpkAutosuggest` has been completely rewritten to make it more accessible for screen reader users.
+
+## History and rationale
+
+The current base library is `react-autosuggest` which is no longer supported and is known to have a11y issues. We had two options: take over ownership of this library or write our own implementation.
+
+Banana has been using a custom implementation built using `downshift`'s `useCombobox` hook since November 2022 which gives screen reader users a better experience. The main improvement is the feedback when results are rendered.
+
+## Migrating
+
+* the component now accepts a ref instead of a ref callback
+
+**Before**
+
+```js
+autosuggestRef: typeof BpkAutosuggest;
+
+<BpkAutosuggest
+  ref={(bpkAutosuggest) => { this.autosuggestRef = bpkAutosuggest; }}>
+
+```
+**Replacement**
+
+
+```js
+autosuggestRef: React.RefObject<HTMLInputElement>;
+
+this.autosuggestRef = React.createRef<HTMLInputElement>();
+<BpkAutosuggest
+  ref={this.autosuggestRef}>
+
+```
+* aria strings are required:
+    * `getA11yResultsMessage` is a callback which returns the number of results being rendered. This should be used to pass translations for single, multiple and no results returned
+    * `ariaLabels` requires a `resultsList` string which will be appended to the `<ul>`
+
+* `renderInputComponent` is no longer a prop. The component renders a BpkInput component, any extension or modification of this can be done via the `inputProps` prop which is passed into the input or by extending or overriding the styling
+
+**Extending**
+
+```js
+this.theme = {
+  ...BpkAutosuggest.defaultProps.theme,
+  input: `${BpkAutosuggest.defaultProps.theme.input} ${STYLES['fsc-location-input']} `,
```

Comment:
This is also used throughout the microsites quite a bit - https://github.skyscannertools.net/search?q=BpkAutosuggest.defaultProps.theme&type=code

Reply:
Exposing the default props was allowing users to override it. I can remove this though. I only added it when I found flights search controls was taking the default suggestionsList styling and adding to it - https://github.skyscannertools.net/web/flights-search-controls/blob/main/src/components/LocationSelector/LocationSelector.tsx#L299

Reply:
Oh you're right, I didn't notice the `theme` prop hidden there at the bottom of the API table - https://github.com/moroshko/react-autosuggest#theme-prop 😞 no, you're totally right to have added given it's being used, I just didn't realise you could override this because I didn't see the prop in the table. That's why I hate opening the API like this 🙃
I do still think we should remove this though given it's only 6 projects using it and we don't want people to have custom designs so we can add this in the migration guide as removed!